### ISSUE TITLE
ADE20K batchsize

### ISSUE THREAD

**Issue by RainHxj** [raoyongming/DenseCLIP#5]
For ADE20K dataset, the batchsize is 32 (4*8gpus) ?

**Comment by raoyongming** [comment]
@RainHxj Yes, we use 8 gpus in our experiments. The global batch size is 32.

**Comment by RainHxj** [comment]
Thanks for your reply. The text encoder is frozen in the training phase, which is implemented by lr=0.0. Compared with no grad, does this method affect the training speed?

**Comment by raoyongming** [comment]
Although the text encoder is fixed during training, we still need to compute its gradients to update the learnable context (the input of the text encoder). Therefore, the `torch.no_grad()` mode cannot be used to accelerate the training.

**Comment by RainHxj** [comment]
Thanks. 
![image](https://user-images.githubusercontent.com/24266484/146783500-cbbb1e36-a296-4264-b560-b79c55d577c5.png)
Does the red box denote using a fixed prompt, such as "a photo of"?

**Comment by raoyongming** [comment]
We use learnable context in this case (Eq. 3 in our case). Using the hard-craft prompt (a photo of `class name`) will lead to a slightly worse result.

**Comment by RainHxj** [comment]
![image](https://user-images.githubusercontent.com/24266484/146786639-170fcac9-7444-440f-b112-05043777ba11.png)
 Thank you. There is a typo. In Table2 clip with prompt,+3.5 and in corresponding text, +2.5

**Comment by RainHxj** [comment]
Thanks for your reply.
